### PR TITLE
967: Update style guide - Part 2

### PR DESCRIPTION
**Pull request by kael89**
### Issue #:
https://github.com/beyondessential/tupaia-backlog/issues/967

Eslint autofixes were applied in `meditrak-server`, `web-config-server`, `web-frontend` - no regression risk

**Still TODO:**
* Use the new TypeScript style guide in `indicators`
* Run the new lint rules in `meditrak-app`
* Update and run the new lint rules in `packages/database/src/migrations`

### CODE REVIEW

File: packages/meditrak-server/src/tests/questionHooks.test.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import { expect } from 'chai';
 
-import { generateTestId, upsertDummyRecord } from '@tupaia/database';
+import { buildAndInsertSurveys, generateTestId, upsertDummyRecord } from '@tupaia/database';
```
Comment (kael89) [inline packages/meditrak-server/src/tests/questionHooks.test.js:3]: This is a manual change - but it's in a test file

File: packages/web-config-server/src/tests/apiV1/dataBuilders/table/tableOfDataValues/tableOfDataValues.fixtures.js [hunk 2]
```diff
@@ -22,7 +22,6 @@ export const DATA_ELEMENTS = {
   CD5: { code: 'CD5', name: 'CVD Risk: Green Female' },
   CD6: { code: 'CD6', name: 'CVD Risk: Green Male' },
   CD7: { code: 'CD7', name: 'CVD Risk: Red Female' },
-  CD8: { code: 'CD8', name: 'CVD Risk: Red Male' },
   CD8: { code: 'CD_Description', name: 'CD Description' },
```
Comment (kael89) [inline packages/web-config-server/src/tests/apiV1/dataBuilders/table/tableOfDataValues/tableOfDataValues.fixtures.js:25]: Another manual change in a test file - this was a duplicate key

File: packages/web-frontend/config/webpack.config.prod.js [hunk 3]
```diff
@@ -167,42 +167,38 @@ module.exports = {
       // in the main CSS file.
       {
         test: /\.css$/,
-        loader: ExtractTextPlugin.extract(
-          Object.assign(
+        loader: ExtractTextPlugin.extract({
+          fallback: require.resolve('style-loader'),
+          use: [
```
Comment (kael89) [inline packages/web-frontend/config/webpack.config.prod.js:172]: This refactoring was done by ESLint

File: packages/web-frontend/src/tests/selectors/orgUnitSelectors.test.js [hunk 4]
```diff
@@ -1,244 +1,258 @@
...
-  });
-});
+/**
+ * Tupaia Web
+ * Copyright (c) 2020 Beyond Essential Systems Pty Ltd.
+ * This source code is licensed under the AGPL-3.0 license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import {
```
Comment (kael89) [inline packages/web-frontend/src/tests/selectors/orgUnitSelectors.test.js:8]: Again, ESLint formatted this file (I believe just added blank lines between test cases)

File: packages/web-frontend/src/tests/selectors/selectors.test.fixtures.js [hunk 5]
```diff
@@ -1,86 +1,86 @@
...
-      { code: 'explore', data: 4, bounds: 'BOUNDS_1' },
-      { code: 'covidau', data: 6, bounds: 'BOUNDS_2' },
-    ],
-  },
-  routing: {
-    pathname: '/explore/TO/UNFPA',
-    search: { MEASURE: 'abc%20123' },
-  },
-};
+export const state = {
```
Comment (kael89) [inline packages/web-frontend/src/tests/selectors/selectors.test.fixtures.js:1]: No idea why this shows up in the diff, the files looks identical 🤔

File: packages/meditrak-server/src/devops/CreateThumbnail/CreateThumbnail.js [hunk 6]
```diff
@@ -1,33 +1,35 @@
+/* eslint-disable no-console */
```
Comment (edmofro) [inline packages/meditrak-server/src/devops/CreateThumbnail/CreateThumbnail.js:1]: Lambda may not support es6, I guess that's why this was all `var`

Comment (kael89) [reply packages/meditrak-server/src/devops/CreateThumbnail/CreateThumbnail.js:1]: We use Node.js 8.10 for this Lambda and full `const` support was added in Node 6.4, but I agree that it's probably not worth the risk changing this!

Comment (kael89) [reply packages/meditrak-server/src/devops/CreateThumbnail/CreateThumbnail.js:1]: Have reverted code changes to minimise regression risk... Just added some eslint directives here

File: packages/web-frontend/config/webpack.config.prod.js (hunk 3 above)
Comment (edmofro) [reply packages/web-frontend/config/webpack.config.prod.js:172]: Hopefully there's no reason the webpack config can't handle some modern es6 features?

File: packages/web-frontend/src/tests/selectors/selectors.test.fixtures.js (hunk 5 above)
Comment (edmofro) [reply packages/web-frontend/src/tests/selectors/selectors.test.fixtures.js:1]: Could have been using a different line break format? No idea...

**Review by edmofro — CHANGES_REQUESTED** [review]
Awesome work @kael89!

I have a few comments. I also had another question - why were we able to take away the eslint disables on the import cycles?

File: packages/meditrak-server/package.json [hunk 7]
```diff
@@ -71,5 +71,8 @@
   "bugs": {
     "url": "https://github.com/beyondessential/tupaia/issues"
   },
-  "homepage": "https://github.com/beyondessential/tupaia"
+  "homepage": "https://github.com/beyondessential/tupaia",
+  "eslintConfig": {
+    "ignorePatterns": "src/database/**"
```
[the remark this answers is not in the document]
Comment (kael89) [reply packages/meditrak-server/package.json:76]: Oops nice catch - I got this pattern from `web-config-server`, where `src/database` contains just migrations, but this is not true in `meditrak-server`. I'll change that

👍 for deleting the migrations folder! Can do in a following PR, if you're keen

File: packages/web-config-server/.eslintrc [hunk 8]
```diff
@@ -1,5 +1,6 @@
 {
   "plugins": ["module-resolver"],
+  "ignorePatterns": "src/database/**",
```
[the remark this answers is not in the document]
Comment (kael89) [reply packages/web-config-server/.eslintrc:3]: Happy with either approach, and for now I am in favour of deleting the folder completely!

File: packages/meditrak-server/package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (edmofro) [reply packages/meditrak-server/package.json:76]: Yep I'm keen

**Comment by kael89** [comment]
About the import cycle disables, the eslint error is still there. I just think that there should be ways to properly break the dependency cycle, instead of hiding it with an `eslint-disable`

File: packages/meditrak-server/package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (kael89) [reply packages/meditrak-server/package.json:76]: Removing this config entirely, since the migrations folder (which we wanted to exclude) will be deleted in the next PR

**Comment by edmofro** [comment]
Oh nice - I agree with that call!

File: packages/meditrak-server/src/routes/getSocialFeed.js [hunk 9]
```diff
@@ -1,7 +1,10 @@
 /**
  * Tupaia MediTrak
  * Copyright (c) 2017 Beyond Essential Systems Pty Ltd
- **/
+ */
+
+/* eslint no-underscore-dangle: ['error', { 'allow': ['_or_'] }] */
```
[the remark this answers is not in the document]
Comment (kael89) [reply packages/meditrak-server/src/routes/getSocialFeed.js:6]: Do you want me to remove the eslint directive here, so that the error shows and perhaps serves as a hint for change?

File: packages/meditrak-server/src/routes/importSurveys/importSurveys.js [hunk 10]
```diff
@@ -1,7 +1,9 @@
 /**
  * Tupaia MediTrak
  * Copyright (c) 2017 Beyond Essential Systems Pty Ltd
- **/
+ */
+
+/* eslint no-underscore-dangle: ['error', { 'allow': ['_conjunction'] }] */
```
[the remark this answers is not in the document]
Comment (kael89) [reply packages/meditrak-server/src/routes/importSurveys/importSurveys.js:6]: True - I will remove this directive

File: packages/meditrak-server/src/routes/getSocialFeed.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (edmofro) [reply packages/meditrak-server/src/routes/getSocialFeed.js:6]: Yep. Ooorr...could be a very easy change as part of this PR

File: packages/web-frontend/config/webpack.config.prod.js (hunk 3 above)
Comment (kael89) [reply packages/web-frontend/config/webpack.config.prod.js:172]: Oops it seems that this may not be the case! This stack overflow thread is about how to do that: https://stackoverflow.com/questions/31903692/how-can-i-use-es6-in-webpack-config-js

I will revert those changes. Great catch, that was a bit careless on my side!!

Comment (kael89) [reply packages/web-frontend/config/webpack.config.prod.js:172]: It's such a pity that ES6 rules can't be disabled automatically by setting `{ env: { es6: false } }`... 😢 

https://github.com/eslint/eslint/issues/3686

Comment (edmofro) [reply packages/web-frontend/config/webpack.config.prod.js:172]: Yeah damn, that would be a great feature

**Review by edmofro — APPROVED** [review]
Approved with one extra commit from @edmofro ;-)

**Comment by kael89** [comment]
Merging in `dev` as dev-related
